config/config.php

```diff
@@ -12,24 +12,15 @@
  * Obviously, if you use closures in your config you can't cache it.
  */
 
-$cachedConfigFile = 'data/cache/app_config.php';
+$configManager = new ConfigManager([
+    // Load config from all modules first
+    new PhpFileProvider('src/*/Config/{{,*.}global,{,*.}local}.php'),
```

Comment:
At first glance this looks like a great idea (no need to register each module separately), but after giving it some thoughts I see a few potential problems:

* all modules would have to follow the same convention (configuration in PHP file, under `src/ModuleName/Config/` directory). You also have to be aware of your modules' directory structure. I suggest using config provider classes instead (like in my initial example).
* local overrides (`*.local.php`) should be placed only in `config/autoload/` directory, otherwise you could end up with multiple local files in different locations.
* using wildcard doesn't let you specify order of modules to be loaded.
* finally, with wildcard it is not explicit which modules are actually loaded (I guess this is matter of personal preference though).